pilot/pkg/model/telemetry.go

```diff
@@ -462,41 +485,48 @@ func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig, mode tpb.
 	var loggingFilter *tpb.AccessLogging_Filter
 	providerNames := mesh.GetDefaultProviders().GetAccessLogging()
 	for _, m := range logs {
-		names := getProviderNames(m.Providers)
-		if len(names) > 0 {
-			providerNames = names
+		names := sets.New()
+		for _, p := range m.Logging {
+			subProviders := getProviderNames(p.Providers)
+			names.InsertAll(subProviders...)
+
+			if p.Filter != nil {
+				loggingFilter = p.Filter
+			}
 		}
 
-		if m.Filter != nil {
-			loggingFilter = m.Filter
+		if len(names) > 0 {
+			providerNames = names.SortedList()
```

Comment:
nit: if we are just adding this to the inScopeProviders do we even need to sort it? Or convert it out of a set?